exporter/exporterhelper/README.md

```diff
@@ -1,28 +1,101 @@
+the default settings should not be modified.
 
-- `retry_on_failure`
-  - `enabled` (default = true)
-  - `initial_interval` (default = 5s): Time to wait after the first failure before retrying; ignored if `enabled` is `false`
-  - `max_interval` (default = 30s): Is the upper bound on backoff; ignored if `enabled` is `false`
-  - `max_elapsed_time` (default = 120s): Is the maximum amount of time spent trying to send a batch; ignored if `enabled` is `false`
 - `sending_queue`
-  - `enabled` (default = true)
-  - `num_consumers` (default = 10): Number of consumers that dequeue batches; ignored if `enabled` is `false`
-  - `queue_size` (default = 5000): Maximum number of batches kept in memory before data; ignored if `enabled` is `false`;
+  - `enabled` (default = `true`)
+  - `num_consumers` (default = `10`): Number of consumers that dequeue batches;
+    ignored if `enabled` is `false`.
+  - `queue_size` (default = `5000`): Maximum number of batches kept in memory
+    before data; ignored if `enabled` is `false`.
   User should calculate this as `num_seconds * requests_per_second` where:
-    - `num_seconds` is the number of seconds to buffer in case of a backend outage
+    - `num_seconds` is the number of seconds to buffer in case of a backend outage.
     - `requests_per_second` is the average number of requests per seconds.
-- `resource_to_telemetry_conversion`
-  - `enabled` (default = false): If `enabled` is `true`, all the resource attributes will be converted to metric labels by default.
-- `timeout` (default = 5s): Time to wait per individual attempt to send data to a backend.
 
-The full list of settings exposed for this helper exporter are documented [here](factory.go).
+### Retry
+
+These settings modify retry. It is recommended to keep retry enabled so data is
+not lost during export in case of error or timeout. If out of order data is a
+concern for the configured destination, disabling retry may be desirable.
+
+> On permanent errors, data is dropped and not retried. On partial error, only
+> non-exported data is retried.
+
+- `retry_on_failure`
+  - `enabled` (default = `true`)
+  - `initial_interval` (default = `5s`): Time to wait after the first failure
+    (which includes a `timeout`) before retrying; ignored if `enabled` is
+    `false`.
+  - `max_interval` (default = `30s`): Is the upper bound on backoff; ignored if
+    `enabled` is `false`.
+  - `max_elapsed_time` (default = `120s`): Is the maximum amount of time spent
+    trying to export data; ignored if `enabled` is `false`. The setting is only
+    checked during a retry backoff operation which means the total time between
+    first attempting to send data to a destination and the time at which
+    `retry_on_failure` stops retrying may be more than `max_elapsed_time`. This
+    setting is completely independent of `timeout`.
+
+Retry happens when data fails to be accepted by the configured destination.
```

Comment:
I would also mention, that not all of the errors cause retries. Some of them are permanent and returned immediately